### ISSUE TITLE
Code Editor 2: Show Pre-Create Event along with the rest of the Object Events, or make a new Pre-Create Object Event

### ISSUE THREAD

**Issue by GameDevTosh** [YoYoGames/GameMaker-Bugs#6496]
### Is your feature request related to a problem?

Seeing how the pre-create seems to be an actual event, which is where variable definitions are set, I think it would be great to have the variables also listed in code for us to edit along with the other events, OR to have the pre-create event itself available for editing. This is so all code related to the object can be in one single place, rather than having a very important event elsewhere that we have to edit separately.

### Describe the solution you'd like

Either a new object event called Pre-Create where we can set our own variables before the create event, or to list all the variables we set in variable definitions also in the event list with all the rest of the events and code for the object to edit. It can work similarly to the Visual to GML conversions if need be for certain things, as long as we can edit things in the code editor.

### Describe alternatives you've considered

Making the variable definitions window optional, and having a pre-create event we can set things to ourselves.

### Additional context

Having the pre-create listed with the other events to edit here:
![pre-create](https://github.com/YoYoGames/GameMaker-Bugs/assets/85319605/4e92c672-5995-4b32-a422-d37b17062192)

**Comment by gnysek** [comment]
Yeah, I would love to see preview of "variable definitions" as code, and I totally don't need to be editable, can be even displayed as a not editable block comment, above "create".

**Comment by DragoniteSpam** [comment]
when there's an error in a Variable Definition and the runtime blames PreCreate, it's hard to figure out what that means without going on the internet and asking

there should either be a PreCreate "event" with a note over it saying "editable in variable definitions" so that people who see an error like that know where to look, or possibly the PreCreate event itself should be a Variable Definition inspector to make it easy for people to find from within CE2

**Comment by gnysek** [comment]
Also, in case of children, it might be even less noticeable, that parent have Variable Definitions.

**Comment by yanako-rpgs** [comment]
This would be incredible, I was asking for something very similar a couple of months ago

**Comment by thennothinghappened** [comment]
Personally I'd love for variable definitions to be presented in-editor. They'd end up taking the form of a stricter code editor window where it won't let you hit save if you try to write something that isn't an assignment. There'd have to be a little bit of smartness for the IDE to get the type correct for variables so it works right with the GUI's representation (vs. being all `expression`, which honestly I wouldn't mind.)
